src/routes/accounts.py

```
@@ -1,25 +1,318 @@
+
+        await db.commit()
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid email or token."
+        )
+
+    try:
+        user._hashed_password = hash_password(data.password)
+        await db.delete(reset_token)
+        await db.commit()
+        return {"message": "Password reset successfully."}
+    except Exception:
+        await db.rollback()
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="An error occurred while resetting the password.",
+        )
+
+
+@router.post(
+    "/login/",
+    response_model=UserLoginResponseSchema,
+    status_code=status.HTTP_201_CREATED,
+)
+async def login(
+    data: UserLoginRequestSchema,
+    db: AsyncSession = Depends(get_db),
+    jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager),
+    settings: BaseAppSettings = Depends(get_settings),
+):
+
+    result = await db.execute(
+        select(UserModel).where(UserModel.email == data.email.lower())
+    )
+    user = result.scalars().first()
+
+    if not user or not verify_password(data.password, user._hashed_password):
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid email or password.",
+        )
+
+    if not user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="User account is not activated.",
+        )
+
+    token_data = {"user_id": cast(int, user.id), "email": user.email}
+    access_token_value = jwt_manager.create_access_token(data=token_data)
+    refresh_token_value = jwt_manager.create_refresh_token(data=token_data)
+
+    try:
+        refresh_token = RefreshTokenModel.create(
+            user_id=user.id,
+            days_valid=settings.LOGIN_TIME_DAYS,
+            token=refresh_token_value,
+        )
+        db.add(refresh_token)
+        await db.commit()
+    except Exception:
```

The DB write try/except around refresh token storage catches all Exceptions. Consider catching `SQLAlchemyError` specifically (or re-raising non-DB exceptions) to avoid masking unexpected errors and to express intent more clearly. Tests that patch `AsyncSession.commit` raise `SQLAlchemyError`, so behavior will remain correct but narrowing the exception is preferable.